### ISSUE TITLE
Workflow is committing artifacts to default branch

### ISSUE THREAD

**Issue by awvwgk** [fortran-lang/fpm-registry#49]
Our current workflow for building the package index is committing it to the default branch. Instead a separate branch or an upload of the package index seems preferable to me. What do you think?

**Comment by LKedward** [comment]
Yes I agree, separating the upload of the generated index would be a good idea.
